File: app/src/main/java/com/bank/testeandroidv2/RetrofitService.java
package com.bank.testeandroidv2;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitService {

    private static final String URL = "https://bank-app-test.herokuapp.com/api/";

    private static Retrofit retrofit;
    private static OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder();

    private static OkHttpClient.Builder getHttpClientBuilder() {
        httpClientBuilder
                .connectTimeout(1, TimeUnit.MINUTES)
                .readTimeout(20, TimeUnit.SECONDS)
                .writeTimeout(1, TimeUnit.MINUTES)
                .addInterceptor(chain -> {
                    Request.Builder requestBuilder = chain.request().newBuilder();
                    requestBuilder.addHeader("Content-Type", "application/x-www-form-urlencoded");
                    requestBuilder.addHeader("Accept", "application/json");
                    return chain.proceed(requestBuilder.build());
                });
        return httpClientBuilder;
    }

    private static OkHttpClient httpClient = getHttpClientBuilder().build();

    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new Retrofit
                    .Builder()
                    .client(httpClient)
                    .baseUrl(URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/br/com/cauejannini/testesantander/commons/integracao/RetrofitService.kt
package br.com.cauejannini.testesantander.commons.integracao

import android.content.Context
import br.com.cauejannini.testesantander.commons.integracao.NetworkUtils
import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.io.IOException

class RetrofitService() {

    val baseUrl = "https://bank-app-test.herokuapp.com/api/"

    val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(baseUrl)
        .addConverterFactory(GsonConverterFactory.create())
        .client(buildClient())
        .build()

    fun <T> createService(serviceClass: Class<T>): T {

        return retrofit.create(serviceClass)
    }

    fun buildClient() : OkHttpClient {

//        val connectionCheckerInterface = Interceptor { chain ->
//
//            if (!NetworkUtils.isConnected(context)) {
//               throw IOException("Você parece estar offline")
//            }
//            chain.proceed(chain.request().newBuilder().build())
//
//        }

//        val authInterceptor = Interceptor { chain ->
//
//            val apiKey = BuildConfig.API_KEY
//
//            var request = chain.request()
//            val url: HttpUrl = request.url().newBuilder()
//                .addQueryParameter("access_key", apiKey)
//                .build()
//            request = request.newBuilder().url(url).build()
//
//            chain.proceed(request)
//
//        }

        return OkHttpClient.Builder()
//            .addInterceptor(connectionCheckerInterface)
//            .addInterceptor(authInterceptor)
            .build()
    }

}